packages/db/src/lib/execution.ts

```diff
@@ -110,3 +147,49 @@ export const getWorkflowNameIntersectFromGranuleIds = async (
   return aggregatedWorkflowCounts
     .map((workflowCounts: { workflow_name: string }) => workflowCounts.workflow_name);
 };
+
+/**
+ * Get cumulus IDs for list of executions
+ *
+ * @param {Knex | Knex.Transaction} knexOrTransaction -
+ *  DB client or transaction
+ * @param {Array<Object>} executions - array of executions
+ * @param {Object} [executionPgModel] - Execution PG model class instance
+ * @returns {Promise<number[]>}
+ */
+export const getApiExecutionCumulusIds = async (
+  knexOrTransaction: Knex | Knex.Transaction,
+  executions: Array<{ arn: string }>,
+  executionPgModel = new ExecutionPgModel()
+) => {
+  const executionCumulusIds: Array<number> = await Promise.all(executions.map(async (execution) =>
+    await executionPgModel.getRecordCumulusId(knexOrTransaction, {
+      arn: execution.arn,
+    })));
+  return [...new Set(executionCumulusIds)];
+};
+
+/**
+ * Get cumulus IDs for all granules associated to a set of executions
+ *
+ * @param {Knex | Knex.Transaction} knexOrTransaction -
+ *  DB client or transaction
+ * @param {Array<Object>} executions - array of executions
+ * @param {Object} [granulesExecutionsPgModel]
+ *   Granules/executions PG model class instance
+ * @returns {Promise<number[]>}
+ */
+export const getApiGranuleExecutionCumulusIdsByExecution = async (
+  knexOrTransaction: Knex | Knex.Transaction,
+  executions: Array<{ arn: string }>,
+  executionPgModel = new ExecutionPgModel(),
+  granulesExecutionsPgModel = new GranulesExecutionsPgModel()
+): Promise<Array<number>> => {
+  const executionCumulusIds = await getApiExecutionCumulusIds(
+    knexOrTransaction, executions, executionPgModel
+  );
+  const granuleCumulusIds = await granulesExecutionsPgModel
+    .searchByExecutionCumulusIds(knexOrTransaction, executionCumulusIds);
+
+  return granuleCumulusIds;
+};
```

Comment:
Methods to support retrieving executions by granule ID was added to retrieve associated executions for a given granule